Helpers/Eligibility.php

```diff
@@ -33,22 +33,18 @@
 use Alma\MonthlyPayments\Model\Data\PaymentPlanEligibility;
```

Comment:
according to this [sonar advice](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=25&id=alma_alma-monthlypayments-magento2&open=AX9P_h7dsF4NpWIA9BXd), I think `EligibilityHelper` is more than just an eligibility Helper

This service has to many properties in my opinion (`$message`, `$eligible`, `$quote`, `$alreadyLoaded`, `$currentFeePlans`) => we have here a mix between a service and a storage object... maybe you can create an Eligiblity state object if you really need to store all this informations and manipulate the state object properties instead of store them into the helper class.

Following methods should be moved in another service => (pseudo suggestion in parenthesis):

* `hasEnabledPaymentPlansInBo()` and all `.*InBo` methods => (a plan config helper)
* `getEnabledConfigPaymentPlans()` => (maybe the same plan config helper)
* `getPaymentType($planKey)` & `buildType()` => (payment type helper) I think the planKey used here is built into `Config\PaymentPlans\PaymentPlanConfig::key()` and this `key()` method could be extracted in the new helper too
* `getEligibilityQuote()` => redondant with `QuoteHelper` (maybe property `$quote` should be stored in this class)
* `checkItemsTypes` that can be renamed as `hasExcludedProduct` => `QuoteHelper` (this is not an Eligibility dependency)
* `getFormattedPrice` => a concrete class `PricingHelper`